src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -15,8 +15,10 @@ internal sealed partial class JsonClassInfo
         // The length of the property name embedded in the key (in bytes).
         private const int PropertyNameKeyLength = 6;
 
-        private readonly List<PropertyRef> _property_refs = new List<PropertyRef>();
-        private readonly List<PropertyRef> _property_refs_sorted = new List<PropertyRef>();
+        private readonly List<PropertyRef> _propertyRefs = new List<PropertyRef>();
+
+        // Cache of properties by first ordering. Use an array for highest performance.
+        private volatile PropertyRef[] _propertyRefsSorted = null;
```

Comment:
Why does this need to be `volatile`?

Reply:
Multiple threads may attempt to overwrite the same field, so this makes sure the value is set atomically across word boundaries.